storage/driver/net/src/main/java/com/emc/mongoose/storage/driver/net/base/pool/BasicMultiNodeConnPool.java

```diff
@@ -122,6 +125,18 @@ private Channel connect() {
 			Loggers.MSG.debug("New connection to \"{}\"", selectedNodeAddr);
 			try {
 				conn = connect(selectedNodeAddr);
+				conn.closeFuture().addListener((ChannelFutureListener) future -> {
+					final String nodeAddr = future.channel().attr(ATTR_KEY_NODE).get();
+
+					synchronized (connsCountMap) {
+						connsCountMap.put(nodeAddr, connsCountMap.get(nodeAddr) - 1);
+					}
+
+					//concurrencyThrottle.release() can be called twice in case of close() is called after a successful release()
+					//so we need to keep track of available permits
+					if (concurrencyThrottle.availablePermits() < concurrencyLevel)
```

Comment:
Hi Evgeny. Why not just to put this to the synchronized block above?  Doesn't this protect us of recurrent invocation?